test/test_svi.py

```diff
@@ -67,3 +69,27 @@ def body_fn(i, val):
     opt_state, rng = fori_loop(0, 300, body_fn, (opt_state, rng_train))
     params = get_params(opt_state, rng, guide_args=())
     assert_allclose(params['loc'], true_coef, atol=0.05)
+
+
+def test_elbo_dynamic_support():
+    x_prior = dist.TransformedDistribution(
+        dist.Normal(), [AffineTransform(0, 2), SigmoidTransform(), AffineTransform(0, 3)])
+    x_guide = dist.Uniform(0, 3)
+
+    def model():
+        sample('x', x_prior)
+
+    def guide():
+        sample('x', x_guide)
+
+    opt_init, opt_update, get_opt_params = optimizers.adam(0.01)
+    # set base value of x_guide is 0.9
+    x_base = 0.9
```

Comment:
Rather than setting this to 0.9, what do you think about validating the elbo computation vmapped over rng and comparing this estimate to the Monte Carlo estimate by drawing n samples from x_guide and computing `np.mean(x_prior.log_prob(samples) - x_guide.log_prob(samples))`? Or maybe we can have these tests separately for different kinds of model/guide pairs.

Reply:
I think we can add another separated test, like `test_kl_dynamic_support()`. Let me try it.

Reply:
I'm happy to merge this and have kl tests in a separate PR as long as existing tests pass.

Reply:
@neerajprad There is nothing change in SVI implementation to handle dynamic support, so having KL test seems not necessary to me. The current test is just to make sure that future changes won't break the logic of elbo.